### PR TITLE
Enabling Travis-CI

### PR DESCRIPTION
**Pull request by abdullahselek**
- Ignore NMAKit, user data and build folders
- Add dependency references for projects
- Share projects schemes for CI
- Add pre-installer and builder
- Add travis configuration file

### CODE REVIEW

File: .ci/buildAll.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+
+# Find paths that contain a xcodeproj
+APP_PROJECTS=$(find "$PWD" -maxdepth 2 -name '*.xcodeproj' -print -prune)
```
Comment (sschuberth) [inline .ci/buildAll.sh]: 1. Please consistently use either single or double quotes. (You may mix them if double quotes are consistently used only where variable interpolation is required, and use single quotes where variable interpolation is not required.)
1. Add `-type d` for clarity?
1. `-print -prune` is not needed here.

File: .ci/buildAll.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+
+# Find paths that contain a xcodeproj
```
Comment (sschuberth) [inline .ci/buildAll.sh]: "[...] contain an xcodeproj directory."

File: .ci/buildAll.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+APP_PROJECTS=$(find "$PWD" -maxdepth 2 -name '*.xcodeproj' -print -prune)
+SDK=iphonesimulator10.3
+DESTINATION="OS=10.3,name=iPhone SE"
+
+for APP_PATH in $APP_PROJECTS; do
+	cd "$APP_PATH"
+	PROJECT=$(find "$PWD" -name '*.xcodeproj' -print -prune)
```
Comment (sschuberth) [inline .ci/buildAll.sh]: Why run `find` again? Don't you just want `.`?

File: .ci/preInstall.sh [hunk 4]
```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
+
+#download the HERE SDK
```
Comment (sschuberth) [inline .ci/preInstall.sh]: `# Download the HERE SDK.`

File: .ci/preInstall.sh [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+wget "$HERE_SDK_URL" -q -O 'HERE_SDK.tar.gz'
+tar -xzf 'HERE_SDK.tar.gz'
+rm 'HERE_SDK.tar.gz'
+
+# Find paths that contain a xcodeproj
+APP_PROJECTS=$(find "$PWD" -maxdepth 2 -name '*.xcodeproj' -print -prune)
```
Comment (sschuberth) [inline .ci/preInstall.sh]: Some comment from `buildAll.sh` apply.

File: .ci/preInstall.sh [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+for APP_PATH in $APP_PROJECTS; do
+    PROJECT_DIR=$(dirname "${APP_PATH}")
```
Comment (sschuberth) [inline .ci/preInstall.sh]: Curly braces around variable name not required here.

**Comment by artemnikitin** [comment]
Maybe it makes sense to add a badge to the readme file as well?

File: .ci/buildAll.sh (hunk 2 above)
Comment (abdullahselek) [inline .ci/buildAll.sh]: Done

File: .ci/buildAll.sh (hunk 1 above)
Comment (abdullahselek) [inline .ci/buildAll.sh]: Done

File: .ci/buildAll.sh (hunk 3 above)
Comment (abdullahselek) [inline .ci/buildAll.sh]: The projects names are not the same with folder names so I am getting xcodeproj and than build them

File: .ci/preInstall.sh (hunk 4 above)
Comment (abdullahselek) [inline .ci/preInstall.sh]: Done

File: .ci/preInstall.sh (hunk 6 above)
Comment (abdullahselek) [inline .ci/preInstall.sh]: Done

File: .ci/preInstall.sh (hunk 5 above)
Comment (abdullahselek) [inline .ci/preInstall.sh]: Done

File: .ci/buildAll.sh (hunk 3 above)
Comment (sschuberth) [inline .ci/buildAll.sh]: But what is this line trying to find, files or directories? Could you use `geocoder-and-reverse-geocoder-ios` as an example to explain?

Comment (abdullahselek) [inline .ci/buildAll.sh]: This line is trying to find "GeocoderRevGeocoderApp.xcodeproj" because xcodebuild command needs .xcodeproj

Comment (sschuberth) [inline .ci/buildAll.sh]: But that's the same as `$(basename "$APP_PATH")`, no?

**Comment by abdullahselek** [comment]
@artemnikitin you are right I will add the badge with in another commit

File: .ci/buildAll.sh [hunk 7]
```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
+
+# Find paths that contain an xcodeproj directory.
+APP_PROJECTS=$(find "$PWD" -mindepth 2 -type d -name "*.xcodeproj")
```
Comment (sschuberth) [inline .ci/buildAll.sh]: Now you dropped the `-maxdepth`, but we should have *both*.

File: .ci/buildAll.sh [hunk 8]
```diff
@@ -0,0 +1,10 @@
...
+SDK=iphonesimulator10.3
+DESTINATION="OS=10.3,name=iPhone SE"
+
+for APP_PATH in $APP_PROJECTS; do
+	xcodebuild -project "$APP_PATH" -sdk "$SDK" -destination "$DESTINATION" -configuration Debug ONLY_ACTIVE_ARCH=NO clean build
```
Comment (sschuberth) [inline .ci/buildAll.sh]: Indentation seems to be wrong, should be 4 spaces.

File: .ci/preInstall.sh [hunk 9]
```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
+
+# Download the HERE SDK.
+wget "$HERE_SDK_URL" -q -O 'HERE_SDK.tar.gz'
+tar -xzf 'HERE_SDK.tar.gz'
+rm 'HERE_SDK.tar.gz'
+
+# Find paths that contain an xcodeproj directory.
+APP_PROJECTS=$(find "$PWD" -mindepth 2 -type d -name "*.xcodeproj")
```
Comment (sschuberth) [inline .ci/preInstall.sh]: Again `-maxdepth` missing.

File: .ci/buildAll.sh (hunk 7 above)
Comment (abdullahselek) [inline .ci/buildAll.sh]: Done

File: .ci/buildAll.sh (hunk 8 above)
Comment (abdullahselek) [inline .ci/buildAll.sh]: Done

File: .ci/preInstall.sh (hunk 9 above)
Comment (abdullahselek) [inline .ci/preInstall.sh]: Done

**Pull request merged**